#1 initialize and configure security

rostyslavdiachuk opened the pull request:
Close #1 Initialize and configure security
 
- Initialize SpringBoot project
- Configure Spring Security
- Add simple login endpoint
- Configure JWT login
- Add IDEA config for formatting
- Add TODO for future

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

rostyslavdiachuk committed:
SS-1 Initialize security service
-- Initialize SpringBoot project
-- Configure Spring Security
-- Add simple login endpoint
-- Configure JWT login
-- Add IDEA config for formatting
-- Add TODO for future

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

rostyslavdiachuk committed:
Delete HELP.md

rostyslavdiachuk committed:
SS-1 Initialize security service
-add method body to test
-fix scoping of method in UserController

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

rostyslavdiachuk committed:
SS-1 Initialize security service
- Fix Test for automatic review

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

rostyslavdiachuk committed:
Merge branch 'develop' into feat/SS-1_initialize_and_configure_security

rostyslavdiachuk committed:
update issues (#6)

* Update issue templates

* Add Codacy badge (#4)

* Update issue templates

rostyslavdiachuk committed:
Merge branch 'develop' into feat/SS-1_initialize_and_configure_security

cuteBrick commented on src/main/java/inc/sebek/securityservice/configuration/SecurityConfiguration.java:
```diff
@@ -0,0 +1,47 @@
+    private AuthenticationManager authenticationManager;
+    private SecurityContextRepository securityContextRepository;
+
+    public SecurityConfiguration(final AuthenticationManager authenticationManager,
```
Isn`t a @Autowired on constructor needed?

rostyslavdiachuk replied:
No, it is not needed. When you have only one constructor in Component(Service,Controller, etc.), Spring automaticaly know that it has `@Autowired`.

cuteBrick commented on src/main/java/inc/sebek/securityservice/security/AuthenticationManager.java:
```diff
@@ -0,0 +1,50 @@
+                                                                                    .map(Roles::mapToGrantedAuthority)
+                                                                                    .collect(toList()));
+            return Mono.just(auth);
+        } else {
```
If you put body of else statement first you will not need else statement and code will be cleaner.

rostyslavdiachuk replied:
Done

cuteBrick commented on src/main/java/inc/sebek/securityservice/security/PBKDF2Encoder.java:
```diff
@@ -0,0 +1,39 @@
+    @SneakyThrows
+    @Override
+    public String encode(CharSequence rawPassword) {
+        var encodedPassword = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA512")
```
Magic String

rostyslavdiachuk replied:
Done

cuteBrick commented on src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    private boolean isPasswordCorrect(String rawPassword, String encodedPassword) {
+        return passwordEncoder.encode(rawPassword).equals(encodedPassword);
```
Doesn`t passwordEncoder have dedicated method for this?

rostyslavdiachuk replied:
Good point

rostyslavdiachuk replied:
Done

rostyslavdiachuk committed:
SS-1 Initialize security service
- Remove "magic strings"
- Add contains logic to Roles
- Refactor authentication logic
- Remove redundant method for matching passwords
- Clean up code

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

rostyslavdiachuk committed:
Merge remote-tracking branch 'origin/feat/SS-1_initialize_and_configure_security' into feat/SS-1_initialize_and_configure_security

rostyslavdiachuk committed:
Merge branch 'master' into feat/SS-1_initialize_and_configure_security
